test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -3749,25 +3745,25 @@ def _get_pid(self, inst):
             path = os.path.join(self.pbs_conf['PBS_HOME'], priv, lock)
         rv = self.du.cat(self.hostname, path, sudo=True, logerr=False)
         if ((rv['rc'] == 0) and (len(rv['out']) > 0)):
-            self.pid = rv['out'][0].strip()
+            pid = rv['out'][0].strip()
         else:
-            self.pid = None
-        return self.pid
+            pid = None
+        return pid
 
-    def _update_pid(self, inst):
+    def _validate_pid(self, inst):
         """
-        update pid of given inst
-
+        Get pid and validate
         :param inst: inst to update pid
         :type inst: object
         """
+        pid = None
         for i in range(30):
             live_pids = self._all_instance_pids(inst)
-            inst.pid = self._get_pid(inst)
-            if live_pids is not None and inst.pid in live_pids:
-                return
+            pid = self._get_pid(inst)
+            if live_pids is not None and pid in live_pids:
+                return pid
             time.sleep(1)
-        inst.pid = None
+        return pid
```

Comment:
Prior to this, we set pid to None.  pid here might not be None.  We get here if live_pids is None or pid is not in live_pids.  Do we need to return None here?

Reply:
Yes. Changed it